### DISCUSSION TITLE
packer build fails and REFUSES to work

### DISCUSSION THREAD

**Discussion by CraigCraig-jpeg** [actions/virtual-environments#6194] [General questions] [answered]
so i have been trying to compile these imasges for days now and its EXTREMELEY painful. For windows 2019 the latest release before the pre release gets stuck at this 
`==> vhd: Provisioning with powershell script: /home/craig/IMAGES/windows-server-2019-Alpha/runner-images-win19-20220828.1/images/win/scripts/Installers/Install-WebPlatformInstaller.ps1
    vhd: Downloading WebPlatformInstaller_x64_en-US.msi...
    vhd: Downloading package from: http://go.microsoft.com/fwlink/?LinkId=287166 to path C:\Users\packer\AppData\Local\Temp\WebPlatformInstaller_x64_en-US.msi .
    vhd: Package downloaded successfully in 0.37 seconds
    vhd: Starting Install WebPlatformInstaller_x64_en-US.msi...
    vhd: Installation successful in 2.05 seconds
    vhd: Pester v5.3.3
    vhd:
    vhd: Starting discovery in 1 files.
    vhd: Discovery found 42 tests in 1.24s.
    vhd: Filter 'FullName' set to ('WebPlatformInstaller').
    vhd: Filters selected 1 tests to run.
    vhd: Running tests.
    vhd:
    vhd: Running tests from 'C:\image\Tests\Tools.Tests.ps1'
    vhd: Describing WebPlatformInstaller
    vhd:   [+] WebPlatformInstaller 933ms (703ms|230ms)
    vhd: Tests completed in 4.8s
    vhd: Tests Passed: 1, Failed: 0, Skipped: 0 NotRun: 41
==> vhd: Restarting Machine
==> vhd: Waiting for machine to restart...
==> vhd: A system shutdown is in progress.(1115)
    vhd: pkrvmars5iqkbb2 restarted.
==> vhd: Machine successfully restarted, moving on
==> vhd: Provisioning with Powershell...
==> vhd: Provisioning with powershell script: /home/craig/IMAGES/windows-server-2019-Alpha/runner-images-win19-20220828.1/images/win/scripts/Installers/Install-VS.ps1
    vhd: Downloading Bootstrapper ...
    vhd: Downloading package from: https://aka.ms/vs/16/release/vs_Enterprise.exe to path C:\Users\INSTAL~1\AppData\Local\Temp\vs_Enterprise.exe .
    vhd: Package downloaded successfully in 0.45 seconds
    vhd: Enable short name support on Windows needed for Xamarin Android AOT, defaults appear to have been changed in Azure VMs
    vhd: Starting Install ...
`
For linux its ubunut 20.04 and same thing its the latest release before the pre release
`\runner-images-ubuntu22-20220828.1\images\linux\scripts\installers\google-chrome.sh 
return 1
`
it fails becuase both of these tests
`Describe "Firefox" -Skip:(Test-IsUbuntu22) {
    It "Firefox" {
        "sudo -i firefox --version" | Should -ReturnZeroExitCode
    }

    It "Geckodriver" {
        "geckodriver --version" | Should -ReturnZeroExitCode
    }
}

Describe "Chrome" {
    It "Chrome" {
        "google-chrome --version" | Should -ReturnZeroExitCode
    }

    It "Chrome Driver" {
        "chromedriver --version" | Should -ReturnZeroExitCode
    }
}`
because its gives the error of
`
geckodriver --version not found 
`

i restrated the process and will post the full error for linux 
here is the link for the images :
linux: https://github.com/actions/runner-images/releases/tag/ubuntu22%2F20220828.1
windows: https://github.com/actions/runner-images/releases/tag/win19%2F20220828.1

I did not change anything!

**Comment by al-cheb** [comment, marked as answer]
1. You should always use the `main` branch to build an image
2. Please attach packer logs against `main` branch

**Reply by CraigCraig-jpeg** [reply]
I will run now and try

**Reply by CraigCraig-jpeg** [reply]
`vhd: Provisioning with powershell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/win/scripts/Installers/Install-VS.ps1
    vhd: Downloading Bootstrapper ...
    vhd: Downloading package from: https://aka.ms/vs/16/release/vs_Enterprise.exe to path C:\Users\INSTAL~1\AppData\Local\Temp\vs_Enterprise.exe .
    vhd: Package downloaded successfully in 0.35 seconds
    vhd: Enable short name support on Windows needed for Xamarin Android AOT, defaults appear to have been changed in Azure VMs
    vhd: Starting Install ...
`

So im still stuck on this step for windows, its been running for an hour and 20 min. What is the expected time on this ?

**Reply by CraigCraig-jpeg** [reply]
vhd: Provisioning with powershell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/win/scripts/Installers/Install-VS.ps1 vhd: Downloading Bootstrapper ... vhd: Downloading package from: https://aka.ms/vs/16/release/vs_Enterprise.exe to path C:\Users\INSTAL~1\AppData\Local\Temp\vs_Enterprise.exe . vhd: Package downloaded successfully in 0.35 seconds vhd: Enable short name support on Windows needed for Xamarin Android AOT, defaults appear to have been changed in Azure VMs vhd: Starting Install ...

So im still stuck on this step for windows, its been running for an hour and 20 min. What is the expected time on this ?

**Reply by al-cheb** [reply]
It depends on many factors: VM performance, location, internet connection bandwidth. In our nightly builds this step takes at least 1 hour, but not more than 3 hours.

**Reply by CraigCraig-jpeg** [reply]
From this point i will wait for 2 hours and check what happens, in the mean time im starting the linux build. i kept things the exact same as i  the packer image and its running in north europe. I did not provision a VM before hand and is spun up by the script within packer. How long do you expect it to take based on the defaults and having packer bring up the infrastructure?

**Reply by CraigCraig-jpeg** [reply]
Thank you for the assistance so far

**Reply by al-cheb** [reply]
You can check how long it takes - https://github.visualstudio.com/runner-images/_build

e.g. https://github.visualstudio.com/runner-images/_build/results?buildId=128702&view=logs&j=011e1ec8-6569-5e69-4f06-baf193d1351e&t=8b69240d-5eca-52e7-db53-21813d4d30a5 - 4h15m

**Reply by CraigCraig-jpeg** [reply]
so on ubuntu: 
`
    azure-arm: Running tests from '/imagegeneration/tests/Tools.Tests.ps1'
    azure-arm: Describing Homebrew
    azure-arm:   [+] homebrew 168ms (137ms|31ms)
    azure-arm:  Context Packages
    azure-arm:    [+] zstd 20ms (14ms|6ms)
    azure-arm: Tests completed in 1.05s
    azure-arm: Tests Passed: 2, Failed: 0, Skipped: 0 NotRun: 98
==> azure-arm: Provisioning with shell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/linux/scripts/base/snap.sh
==> azure-arm: Provisioning with shell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/linux/scripts/base/reboot.sh
    azure-arm: Reboot VM
==> azure-arm: Pausing 1m0s before the next provisioner...
==> azure-arm: Provisioning with shell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/linux/scripts/installers/cleanup.sh
    azure-arm: /usr/bin/journalctl
==> azure-arm: Vacuuming done, freed 0B of archived journals from /run/log/journal.
==> azure-arm: Vacuuming done, freed 0B of archived journals from /var/log/journal.
==> azure-arm: Deleted archived journal /var/log/journal/a73dede730094d4ca54100906e0f076d/system@f6340573e5fa4d608d480856661d2cb6-0000000000000001-0005e826581b8b25.journal (25.0M).
==> azure-arm: Deleted archived journal /var/log/journal/a73dede730094d4ca54100906e0f076d/user-1000@4eca7ed4eaaf40588ab86be0346b2b52-0000000000000509-0005e8265a5587e0.journal (8.0M).
==> azure-arm: Deleted archived journal /var/log/journal/a73dede730094d4ca54100906e0f076d/system@f6340573e5fa4d608d480856661d2cb6-0000000000001de7-0005e82749e96f4c.journal (8.0M).
==> azure-arm: Deleted archived journal /var/log/journal/a73dede730094d4ca54100906e0f076d/user-1000@4eca7ed4eaaf40588ab86be0346b2b52-0000000000001ffe-0005e827951fd2ae.journal (8.0M).
==> azure-arm: Vacuuming done, freed 49.0M of archived journals from /var/log/journal/a73dede730094d4ca54100906e0f076d.
    azure-arm: Before: 27837 MB
    azure-arm: After : 30665 MB
    azure-arm: Delta : 2828 MB
==> azure-arm: Provisioning with shell script: /home/craig/WORKS/runner-images-main/runner-images-main/images/linux/scripts/base/apt-mock-remove.sh
==> azure-arm: Provisioning with shell script: /tmp/packer-shell2268113220
    azure-arm: Running SoftwareReport.Generator.ps1 script
==> azure-arm: time="2022-09-08T10:08:43Z" level=warning msg="Failed to decode the keys [\"machine\"] from \"/usr/share/containers/containers.conf\"."
==> azure-arm:  ›   Warning: Our terms of service have changed:
==> azure-arm:  ›   https://dashboard.heroku.com/terms-of-service
==> azure-arm: /imagegeneration/SoftwareReport/SoftwareReport.Generator.ps1 : The term 'zstd' is not recognized as a name of a cmdlet, function, script file, or executable program.
==> azure-arm: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
==> azure-arm: + CategoryInfo          : ObjectNotFound: (zstd:String) [SoftwareReport.Generator.ps1], CommandNotFoundException
==> azure-arm: + FullyQualifiedErrorId : CommandNotFoundException,SoftwareReport.Generator.ps1
==> azure-arm: Provisioner failed with "Script exited with non-zero exit status: 1. Allowed exit codes are: [0]", retrying with 3 trie(s) left
==> azure-arm: Provisioning with shell script: /tmp/packer-shell2778173777
    azure-arm: Running SoftwareReport.Generator.ps1 script
==> azure-arm: time="2022-09-08T10:09:16Z" level=warning msg="Failed to decode the keys [\"machine\"] from \"/usr/share/containers/containers.conf\"."
==> azure-arm: /imagegeneration/SoftwareReport/SoftwareReport.Generator.ps1 : The term 'zstd' is not recognized as a name of a cmdlet, function, script file, or executable program.
==> azure-arm: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
==> azure-arm: + CategoryInfo          : ObjectNotFound: (zstd:String) [SoftwareReport.Generator.ps1], CommandNotFoundException
==> azure-arm: + FullyQualifiedErrorId : CommandNotFoundException,SoftwareReport.Generator.ps1
==> azure-arm: Provisioner failed with "Script exited with non-zero exit status: 1. Allowed exit codes are: [0]", retrying with 2 trie(s) left
==> azure-arm: Provisioning with shell script: /tmp/packer-shell581136908
    azure-arm: Running SoftwareReport.Generator.ps1 script
==> azure-arm: time="2022-09-08T10:09:44Z" level=warning msg="Failed to decode the keys [\"machine\"] from \"/usr/share/containers/containers.conf\"."
==> azure-arm: /imagegeneration/SoftwareReport/SoftwareReport.Generator.ps1 : The term 'zstd' is not recognized as a name of a cmdlet, function, script file, or executable program.
==> azure-arm: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
==> azure-arm: + CategoryInfo          : ObjectNotFound: (zstd:String) [SoftwareReport.Generator.ps1], CommandNotFoundException
==> azure-arm: + FullyQualifiedErrorId : CommandNotFoundException,SoftwareReport.Generator.ps1
==> azure-arm: Provisioner failed with "Script exited with non-zero exit status: 1. Allowed exit codes are: [0]", retrying with 1 trie(s) left
==> azure-arm: Provisioning with shell script: /tmp/packer-shell3619707731
    azure-arm: Running SoftwareReport.Generator.ps1 script
==> azure-arm: time="2022-09-08T10:10:12Z" level=warning msg="Failed to decode the keys [\"machine\"] from \"/usr/share/containers/containers.conf\"."
==> azure-arm: /imagegeneration/SoftwareReport/SoftwareReport.Generator.ps1 : The term 'zstd' is not recognized as a name of a cmdlet, function, script file, or executable program.
==> azure-arm: Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
==> azure-arm: + CategoryInfo          : ObjectNotFound: (zstd:String) [SoftwareReport.Generator.ps1], CommandNotFoundException
==> azure-arm: + FullyQualifiedErrorId : CommandNotFoundException,SoftwareReport.Generator.ps1
==> azure-arm: retry limit reached.
==> azure-arm: Script exited with non-zero exit status: 1. Allowed exit codes are: [0]

==> azure-arm: Step "StepProvision" failed
==> azure-arm: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may
`

**Reply by CraigCraig-jpeg** [reply]
the win build is still running

**Reply by CraigCraig-jpeg** [reply]
any thaughts?

**Reply by al-cheb** [reply]
For Ubuntu it's a well-known issue - https://github.com/actions/runner-images/issues/6200

**Reply by CraigCraig-jpeg** [reply]
should i just remove this step within packer?

**Reply by CraigCraig-jpeg** [reply]
like the files?

**Reply by CraigCraig-jpeg** [reply]
Got it fixed for windows. For linux i removed the steps that were failling and its compiling

**Comment by samus-aran** [comment]
We're locking this discussion because it has not had recent activity and/or other members have asked for more information to assist you but received no response. Thank you for helping us maintain a productive and tidy community for all our members.
